File: app/src/main/java/com/argos/android/opencv/Fragment/ChooseImageDialogFragment.java
package com.argos.android.opencv.Fragment;

import android.app.Dialog;
import android.support.design.widget.BottomSheetDialogFragment;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import com.argos.android.opencv.R;
import com.argos.android.opencv.Adapter.ImageListAdapter;

/**
 * BottomSheet for choosing image. Refer {@link com.argos.android.opencv.Activity.ImageLoadActivity}
 * Add images of resolution 640x480 inside drawables and include them in {@link #images}
 */

public class ChooseImageDialogFragment extends BottomSheetDialogFragment
{
    private RecyclerView recyclerView;
    private ImageListAdapter imageListAdapter;
    private int[] images = {R.drawable.scene_1, R.drawable.scene_2};

    @Override
    public void setupDialog(Dialog dialog, int style)
    {
        View contentView = View.inflate(getContext(), R.layout.choose_image, null);
        dialog.setContentView(contentView);

        recyclerView = (RecyclerView) contentView.findViewById(R.id.image_list);
        imageListAdapter = new ImageListAdapter(images, getArguments().getString("feature"), getContext());

        recyclerView.setAdapter(imageListAdapter);
        recyclerView.setLayoutManager(new LinearLayoutManager(getContext(),LinearLayoutManager.HORIZONTAL, false));
    }
}

File: app/src/main/java/com/argos/android/opencv/fragment/ChooseImageDialogFragment.kt
package com.argos.android.opencv.fragment

import android.app.Dialog
import android.support.design.widget.BottomSheetDialogFragment
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.view.View
import com.argos.android.opencv.R
import com.argos.android.opencv.adapter.ImageListAdapter

/**
 * BottomSheet for choosing image. Refer [com.argos.android.opencv.activity.ImageLoadActivity]
 * Add images of resolution 640x480 inside drawables and include them in [.images]
 */

class ChooseImageDialogFragment : BottomSheetDialogFragment() {
    private var recyclerView: RecyclerView? = null
    private var imageListAdapter: ImageListAdapter? = null
    private val images = intArrayOf(
            R.drawable.scene_1,
            R.drawable.scene_2,
            R.drawable.track_01,
            R.drawable.track_02,
            R.drawable.track_03,
            R.drawable.track_04,
            R.drawable.track_05,
            R.drawable.track_06,
            R.drawable.speed_dreams1,
            R.drawable.speed_dreams3,
            R.drawable.speed_dreams4)

    override fun setupDialog(dialog: Dialog, style: Int) {
        val contentView = View.inflate(context, R.layout.choose_image, null)
        dialog.setContentView(contentView)

        recyclerView = contentView.findViewById<View>(R.id.image_list) as RecyclerView
        imageListAdapter = ImageListAdapter(images, arguments!!.getString("feature")!!, context!!)

        recyclerView!!.adapter = imageListAdapter
        recyclerView!!.layoutManager = LinearLayoutManager(context, LinearLayoutManager.HORIZONTAL, false)
    }
}
